Initialize stroke and geometry when control attaches to the visual tree

tiny-isle opened the pull request:
Adds OnAttachedToVisualTree to ensure _stroke and _geometry are initialized correctly, fixing cases where inherited properties (e.g., Foreground) were not ready during initial load.
<img width="1963" height="442" alt="image" src="https://github.com/user-attachments/assets/7618df47-6646-4ed5-ab78-7942237eb7ce" />

tiny-isle committed:
Initialize stroke and geometry when control attaches to the visual tree

dme-compunet commented:
Thank you for your contribution!

Could you please share a code snippet that demonstrates the issue this PR is addressing?

tiny-isle commented:
Thank you for your reply. I tried to reproduce this issue in a new project, but failed. It's possible that something in my current project is affecting it, but I haven't found the cause yet.

The issue started when I rewrote the control template for ComboBoxItem, replacing the PathIcon with a LucideIcon. I found that the LucideIcon was not working under certain conditions. When debugging with Rider, I observed that the Foreground property had a value when the LucideIcon triggered a Render call, but the stroke was not being created.

Since the Foreground already had a value when the control was attached to the visual tree, I suspect that the Foreground property change event might not have been triggered. Therefore, I added a check for the stroke when the control is attached to the visual tree.
My operation steps involve switching to a different theme, at which point the icon can be rendered normally. However, when the interface is first created and displayed, the icon is not visible.

I have an operation video and a Rider breakpoint video available for reference below.

Finally, I would like to express my sincere gratitude to the author for providing such an excellent and user-friendly icon library.

https://github.com/user-attachments/assets/2068d3eb-3325-4b9c-953a-a6b09a4a85b5

tiny-isle committed:
Merge branch 'dme-compunet:master' into ensure-foreground-exist

dme-compunet commented:
Thank you for your detailed explanation. It’s possible that initializing `_stroke` in `OnPropertyChanged` wasn’t the best idea from the start.
By the way, I’ve encountered issues similar to what you described with text rendering before. Which version of Avalonia are you using?

tiny-isle commented:
Thank you for your detailed analysis.

The Avalonia version I am currently using is 11.3.8
